src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderStateController.php

```diff
@@ -398,4 +411,24 @@ private function getErrorMessages(\Exception $e)
             ),
         ];
     }
+
+    /**
+     * @return bool
+     */
+    protected function isMultistoreAllShopsContextForced(): bool
+    {
+        return $this->get('prestashop.adapter.multistore_feature')->isUsed()
+            && $this->get('prestashop.adapter.shop.context')->isShopContext();
+    }
+
+    /**
+     * @return string
+     */
+    protected function getMultistoreAllShopsContextForcedMessage(): string
+    {
+        return $this->trans(
```

Comment:
Shouldn't be something more generic, will be something only available for ORderState context? or useful for some controllers?

Reply:
The strings is being used in many controllers as you can see in https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/BackupController.php#L87

So far it's been added in each Controller when needed see https://github.com/PrestaShop/PrestaShop/pull/24032

Maybe we'll be able to harmonize the messages when we build https://github.com/PrestaShop/PrestaShop/issues/25016 ?

Reply:
Yes good idea to harmonize it in a new PR.
I'm afraid of duplicate code :sweat_smile: